app/apollo/resolvers/channel.js

```diff
@@ -85,17 +77,21 @@ const channelResolvers = {
       try{
         checkComplexity( queryFields );
 
-        const channel = await models.Channel.findOne({org_id, uuid });
+        const channel = await models.Channel.findOne({org_id, uuid});
         if (!channel) {
           throw new NotFoundError(context.req.t('Could not find the configuration channel with uuid {{uuid}}.', {'uuid':uuid}), context);
         }
+
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating`);
         await validAuth(me, org_id, ACTIONS.READ, TYPES.CHANNEL, queryName, context, [channel.uuid, channel.name]);
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating - authorized`);
+
```

Comment:
- The `validating` log message should be at the beginning of the validation section (i.e. right after the `try {` line).
- Simple checks for valid input should be first
- While the DB query can be before the `validAuth` call, the throwing of NotFoundError should be _after_, to avoid information leakage.
- The validAuth call will need to change it's `[ id, name ]` argument depending on whether the cluster was found in the DB or not.